chapter_post_and_database.asciidoc

```diff
@@ -1213,41 +1176,106 @@ def test_can_save_a_POST_request(self):
+// JAN: I see what you want to show here.
+// Yet, I have one problem with this test.
+// It's using two different levels of abstraction - HTTP client and DB model.
+// In other words, you're testing iplementation details (I'm posting to server and it needs to store this to database).
+// I know it's Django so it's hard to not test implementation details.
+// Nevertheless, I think it's worth to think about it.
+// That's because this is one of the most common pitfalls that I see in tests.
+// Such tests lock you down to specific implemetation.
+// Consequently, you need to change them whenever you change the implementation.
+// For example, let's say that we want TODO list to be temporary thing - it's stored only for 7 days.
+// And we want to use Redis for it due to TTL.
+// We'll need to rewrite such test.
+// That means that this test is not resistant to refactoring (e.g., we'll refactor first and use TTL=100 years).
+// I'd maybe try to follow similar principle as you did in FT.
+// Let's write a test that can't pass just by returning sent data.
+// You used two requests and 2 assertions.
+// I think such test will be much more valuable than mixture of client and DB model.
+// It could look like this:
+
 ////
 ====
 [source,python]
 ----
 def test_can_save_a_POST_request(self):
     response = self.client.post("/", data={"item_text": "A new list item"})
     self.assertContains(response, "A new list item")
-    
+
     response = self.client.post("/", data={"item_text": "Kiss your wife"})
     self.assertContains(response, "A new list item")
     self.assertContains(response, "Kiss your wife")
-    
+
     self.assertTemplateUsed(response, "home.html")
 ----
 ====
 ////
-// JAN: Such test will pass as long as you use something to persist TODO items across requests. To make it pass at this point, you'll need to either spent a lot of code to trick it or use models which is simpler.
+
+// JAN: Such test will pass as long as you use something to persist TODO items across requests.
+// To make it pass at this point, you'll need to either spent a lot of code to trick it or use models which is simpler.
```

Comment:
hey @jangia just to let you know i'm still v intrigued by this suggestion.  i had a first go at it, see notes below, it's not quite working yet but i hope to be able to try again at some point.

one weird thing is that it's actually quite easy to persist things across requests in your tests, you just use a global variable.  and that actually  *more powerful* persistence than django models as far as tests are concerned, because django resets the db between each test, whereas the global variable sits there polluting state across tests for the whole session lol.

so haven't quite found the right approach to the chapter yet but i'll keep trying.